cdap-docs/developer-manual/source/pipelines/developing-plugins/plugin-basics.rst

```diff
@@ -126,3 +126,14 @@ In this example, we have a plugin of type ``batchsource``, named ``MyBatchSource
 This plugin takes two configuration properties. The first is named ``input-path`` and is required.
 The second is named ``cleanOutput`` and is optional. Note that optional configuration fields should
 have their default values set in the no-argument constructor.
+
+Plugin Data types
+=================
+
+Plugins support various data types such as boolean, bytes, double, float, int, long, string, array, enum, map, union
+and record. Temporal data types such as date, time and timestamp are also supported. Date represents number of days
+since epoch. Time represents number of microseconds after midnight. Timestamp represents number of microseconds from
+unix epoch, 1 January 1970 00:00:00.000 UTC.
+
```

Comment:
There should be a code sample for this where we demonstrate how to build a Schema and a StructuredRecord. We should demonstrate how somebody creates these logical types as fields in a schema, and also how they set it in a StructuredRecord. Also need to mention that each logical type has a corresponding physical type that can be fetched through the StructuredRecord.get() method.